js/indicia.datacomponents/idc.core.js

```diff
@@ -1245,6 +1245,45 @@
       if ($('.permissions-filter').length > 0) {
         data.permissions_filter = $('.permissions-filter').val();
       }
+      // recordContext select drop down.
```

Comment:
Could this code be refactored into a separate function, to avoid such a big method? It will also need a few changes if merged with permissionsFilters as suggested.

Reply:
What do you mean by merging with permissionFilters John?

Reply:
Sorry, think I wrote a comment on the PHP review and forgot to save it. I wondered if the new functionality of the recordContext control should actually be merged into the existing permissionsFilters control rather than adding something new. Quite a lot of the functionality is duplicated between the 2 and I don't see the need to have 2 different controls, as long as you can enable/disable which types of filter it allows.

Reply:
Yes I agree with that - I will modify.

Reply:
I've replaced recordContext with an enhanced permissionFilters control as you suggested John. Can you just clarify what code it is that you would like to be lifted into a separate function?